Kwf/View/Helper/HighlightTerms.php

```diff
@@ -147,11 +148,19 @@ public function highlightTerms($terms, $text, array $options = array())
         // highlighting
         $c = 1;
         foreach ($terms as $term) {
-            $ret = preg_replace(
-                "/(^|\W)($term)(\W|$)/i",
-                '$1<span class="highlightTerms highlightTerm'.$c.'">$2</span>$3',
-                $ret
-            );
+            if ($options['highlightWordPartial']) {
+                $ret = preg_replace(
+                    '/(^|\W)('.$term.')(.|$)/i',
+                    '$1<span class="highlightTerms highlightTerm'.$c.'">$2</span>$3',
+                    $ret
+                );
+            } else {
+                $ret = preg_replace(
+                    '/(^|\W)('.$term.')(\W|$)/i',
+                    '$1<span class="highlightTerms highlightTerm'.$c.'">$2</span>$3',
+                    $ret
+                );
+            }
```

Comment:
Since the second and third parameter in preg_replace are the same, don't duplicate the code, just put the pattern itself in the if-condition